File: src/main/java/um/haberes/core/hexagonal/liquidaciones/item/domain/model/ItemVersion.java
package um.haberes.core.hexagonal.liquidaciones.item.domain.model;

import java.math.BigDecimal;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ItemVersion {

    private Long itemVersionId;

    private Long legajoId;

    private Integer anho;

    private Integer mes;

    private Integer codigoId;

    @Builder.Default
    private String codigoNombre = "";

    @Builder.Default
    private BigDecimal importe = BigDecimal.ZERO;
}

File: src/main/java/um/haberes/core/kotlin/model/ItemVersion.kt
package um.haberes.core.kotlin.model

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table
data class ItemVersion(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var itemVersionId: Long? = null,

    var legajoId: Long? = null,
    var anho: Int = 0,
    var mes: Int = 0,
    var codigoId: Int? = null,
    var codigoNombre: String = "",
    var importe: BigDecimal = BigDecimal.ZERO

) : Auditable()
